### ISSUE TITLE
FE learning center mobile and desktop homepage updates

### ISSUE THREAD

**Issue by jenniferlee-dsva** [department-of-veterans-affairs/va.gov-team#15649]
### You can test the FE template update against this test CMS content in preview-prod: http://preview-prod.vfs.va.gov/preview?nodeId=8296 

**AC - New design elements to update/add in the FE - mobile and desktop:** 

- [x]  Additional benefit topic categories- [See FE ux logic](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/tier-2-content-IA-and-design/learning-center-mvp/LC-homepage-requirements.md#browse-by-topics-section)
- [ ] 508 compliant
- [x] Functionality validated by @jenniferlee-dsva before go-live
- [ ]  Mobile and desktop designs and ix validated by @rtwell before go-live
- [ ] ~Transformer updated~ {12/9 UPDATE: we will use manual content process, so these iterations should not need transformer changes. }
 ~Browse by audience - [See FE ux logic](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/tier-2-content-IA-and-design/learning-center-mvp/LC-homepage-requirements.md#browse-by-audience-section)~ Audience update is now a [separate subticket here, #17276](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/17276)

## Desktop 1.1 - fewer than 5 audiences

![VA-RS-Landing-1-1-1-3-audiences.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/b5694c53-97c7-4beb-9cad-15957eec1bb1)

## Desktop 1.1 - 5+ audiences (collapsed/hidden)

![VA-RS-Landing-1-1-2-more-5-audiences.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/81ddf7f1-c714-41b2-805b-8e3e8494eecb)

## Desktop 1.1 -  expanded/show all states

![VA-RS-Landing-1-1-2-expanded.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/6e0bbf2b-8b98-48d7-8a0e-a6452a931c74)

## Mobile 1.1

![VA-RS-Landing-1-1-2-mobile.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/af285fce-062c-46ac-a356-fad1ea7eb118)

**Comment by andaleliz** [comment]
Hey team! [Here are the mobile designs](https://preview.uxpin.com/716fd9c0d105393853b9655f4edd35f11dbca438#/pages/134261285/comment/sitemap?mode=i) and some documentation. Should be pretty straightforward, leveraging existing design system components. 

Is what I've included sufficient for the `load more` behavior, or do you need further documentation and screens?

**Comment by andaleliz** [comment]
We should get confirmation that [this is still the current desktop design](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1588167553516/2773c854e8ff3a725a5e3ee03272b78e8519f78b?wid=0-1591028076023) for devs to work from.  It may get confusing since it's on the board under the v1 designs, and we worked from another part of the board for v1. 

@jenniferlee-dsva can you confirm here please?

**Comment by jenniferlee-dsva** [comment]
Adding a blocked flag here. @rtwell  is updating interim 1.1 desktop design, so we'll upload final final desktop and mobile here so everything is ready for you together at the same time.

**Comment by jenniferlee-dsva** [comment]
I've cleaned up the OG ticket with final mockups and AC. Please be sure to look at the UX interaction logic I link to.

**Comment by youngfreezyVA** [comment]
here is the latest:
https://dsva.slack.com/archives/C0173CAJ7A6/p1606843750459500
i'd suggest breaking the transformers work out into a separate ticket, otherwise would like to bump this up to 13 points instead of 8. ideally we we could also break out the audiences/topics work into separate tickets as well.
cc: @brianalloyd @jenniferlee-dsva

**Comment by brianalloyd** [comment]
@youngfreezyVA definitely will break out transformers into separate ticket. Thanks.

**Comment by jenniferlee-dsva** [comment]
Moving to backlog bc we need content and cms changes, per 12/9 CMS weekly check-in. 

For now, we're switching to manual content process. Too much transformer lift to go out  with FE automated process for now.

**Comment by jenniferlee-dsva** [comment]
@youngfreezyVA  @brianalloyd 

Great idea ^ re: 
>i'd suggest breaking the transformers work out into a separate ticket, 

- In fact, with the interim manual approach we decided on today, there shouldn't be any transformers update for this work. 

>ideally we we could also break out the audiences/topics work into their own tickets as well.

- Agree. In fact, I think we can do the "Browse by topics" section first. The cms data is already available on the RS homepage node for this.   We could do this and unblock the content team from publishing new articles content right away if we did this separately.

**Comment by jenniferlee-dsva** [comment]
@youngfreezyVA @kelsonic @ncksllvn @brianalloyd  - I've separated out the Audience work from this ticket into a [separate subticket here, #17276](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/17276). 

This main ticket can now focus on updating the RS homepage with more benefit category functionality - which should be fairly (?) simple.

**Comment by brianalloyd** [comment]
Thanks @jenniferlee-dsva for the update and separate ticketing to ease the process.  @kelsonic @ncksllvn @youngfreezyVA  can you review please reestimate this ticket to ensure we're aligned on estimations?  Looks like were unblocked on this now as well, so would like to prioritize this ticket if someone has bandwidth.

**Comment by jenniferlee-dsva** [comment]
You can see the complete FE logic here: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content/tier-2-content-IA-and-design/learning-center-mvp/LC-homepage-requirements.md#browse-by-topics-section  (link in the og description)

Tl;dr - I think the only thing that needs to be added with this ticket now are:

1. Remove the carat > symbol in the FE when there is no "Go to all articles" text link. (Rn, an empty carat appears in preview:  http://preview-prod.vfs.va.gov/preview?nodeId=8296 )

2. Change FE display from 2 categories per row to 3 categories per row. 

3. Add `Show more` functionality for when there are more than 6 categories; make this fuctionality hidden - so it shows only when 6+.

3. Add display order of topic categories which is:
 -  `VA account and profile `is always listed first
- `Other topics and questions` is always listed as last category in display order.
 -  The rest of the topic categories  -- when they are published --  are displayed alpha.

**Comment by brianalloyd** [comment]
@youngfreezyVA or @kelsonic if you have bandwidth this is a priority ticket, please review and reestimate this ticket to ensure we're aligned on estimations.  Thanks for the support.

**Comment by kelsonic** [comment]
I guess I should clarify -- is this higher priority than transformers atm? 🙂

**Comment by brianalloyd** [comment]
@kelsonic negative, keep plugging away at transformers.  If any space opens up or if you want a break from all the transforming we could pick this one up.  Also if @youngfreezyVA has any open space this would be a good ticket to pick up as well.

**Comment by jenniferlee-dsva** [comment]
@kelsonic if you are continuing your transformer work, please continue to focus on finishing those. 

I don't think this particular ticket requires any transformer changes. It is however blocking new articles from being published (or to be precise from displaying as published on the homepage).

It would be good to size this however, so we can make sure to complete this in time before code freeze.

**Comment by jenniferlee-dsva** [comment]
@kelsonic - there is no deadline to finish transformer by code freeze. We know already and platform knows that transformer work will continue into January.

**Comment by brianalloyd** [comment]
@kelsonic Please shift priority to this ticket for today, would like to get this unblocked for the content team prior to 'code freeze' as advised.  The empty carat issue is baked into AC on this ticket.  Image shown below;
![image](https://user-images.githubusercontent.com/70410912/102807489-f8414e80-437b-11eb-8f77-37c1cd81183a.png)

**Comment by kelsonic** [comment]
PR is up for #1: 
```
Remove the carat > symbol in the FE when there is no "Go to all articles" text link. (Rn, an empty carat appears in preview: http://preview-prod.vfs.va.gov/preview?nodeId=8296 )
```

https://github.com/department-of-veterans-affairs/vets-website/pull/15455

**Comment by jenniferlee-dsva** [comment]
@kelsonic Thank you! For awareness, @RLHecht. After Kelson's PR above is merged (for empty carat issue), we should be able to publish additional articles listings on the RS homepage. 

Note: this pr is only to address the empty carat. We won't have the other parts of the RS homepage update until after the freeze. But feel free to start adding more categories of articles when they're ready. 

CC @bethpottsVADEPO

**Comment by brianalloyd** [comment]
@ncksllvn can you prioritize this ticket.  @kelsonic completed the carat removable to unblock content publishing however the remaining items on the ticket need to be worked and validated.  Please reach out if you have any questions.  Thanks, Brian.

**Comment by ncksllvn** [comment]
@brianalloyd I am completely lost on this ticket now...I'll try to read through it all again and follow up with questions but right now it's really unclear to me what the action items are.

**Comment by ncksllvn** [comment]
Or @kelsonic maybe you can help me understand?

**Comment by brianalloyd** [comment]
@ncksllvn thanks for the follow up, totally understand.  @kelsonic please fill in the blanks that I miss but it's my understanding that Kelson completed items #1 however #2-4 remain on the ticket: 

2. Change FE display from 2 categories per row to 3 categories per row.
 
3. Add `Show more` functionality for when there are more than 6 categories; make this functionality hidden - so it shows only when 6+.
 
4. Add display order of topic categories which is:

- `VA account and profile `is always listed first
- ` Other topics and questions` is always listed as last category in display order.
- The rest of the topic categories (like Health care, Education, etc. )-- when they are published on RS home-- are displayed alpha.

**Comment by kelsonic** [comment]
Hey @ncksllvn I can grab the rest of this ticket, I had mistakenly thought this ticket was the one where we are adding the new audience sections to the homepage, which I believe required a separate graphql query that then adds that data to the homepage template, that ticket I'd love your guidance on! 🙂

**Comment by brianalloyd** [comment]
Thanks @kelsonic for the follow up, appreciate it.

**Comment by ncksllvn** [comment]
If I follow correctly then the "browse by audience" work is broken out into this issue, https://github.com/department-of-veterans-affairs/va.gov-team/issues/17276. I'll pick up that one if that's okay

**Comment by kelsonic** [comment]
That'd be perfect, thanks @ncksllvn!! 🔥

**Comment by kelsonic** [comment]
Also, random comment but I think a ticket of this size with this many decoupled AC should probably be broken out into separate tickets (just for future tickets), that way we don't end up with an 8 pointer! 😂 🙂

**Comment by kelsonic** [comment]
Quick question about #2: "Change FE display from 2 categories per row to 3 categories per row."

Is this desired?

![image](https://user-images.githubusercontent.com/12773166/103812734-3301e280-501c-11eb-9c78-2325cff5ed76.png)

**Comment by kelsonic** [comment]
Deleted latest comment as I realize I just looked at the design spec wrong 💯

**Comment by kelsonic** [comment]
> Add display order of topic categories which is:
> 
> VA account and profile is always listed first
> Other topics and questions is always listed as last category in display order.
> The rest of the topic categories (like Health care, Education, etc. )-- when they are published on RS home-- are displayed alpha.

So here's the thing 😬 Currently each category is what's called a `paragraph-lists_of_links`, which is used by more than just the homepage. If I format the categories for a particular order on the FE, it will potentially affect all the places we use lists_of_links. Is there any way that we can do the ordering of the categories from the cms side when inputting content?

**Comment by kelsonic** [comment]
PR is up! 🙂 https://github.com/department-of-veterans-affairs/vets-website/pull/15569

**Comment by kelsonic** [comment]
PR merged, will be live following the daily deploy today 🙂

**Comment by jenniferlee-dsva** [comment]
@kelsonic - does this mean after today, the RS home will show 3 categories per row and using that alpha order for benefit topics?

**Comment by jenniferlee-dsva** [comment]
i see your local screenshots in the pR ticket - awesome - thank you!

**Comment by jenniferlee-dsva** [comment]
I see the new categories in the requested alpha order! Thank you @kelsonic  @ncksllvn . 

Functionality also appears in mobile view, but it looks like there's a lot of excess spacing between the search bar and the next section. 

###  Here's what it looks like live in mobile:
![image.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/e1851ce2-7639-4945-be5a-49422fd219d2)

### And here's the spacing shown in @rtwell 's [mobile design:](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/af285fce-062c-46ac-a356-fad1ea7eb118) 

![image.png](https://images.zenhubusercontent.com/5bb2975c143b4d2ee7a35ef6/4b7fbd54-d8f9-404c-83a2-881532df9917)

**Comment by kelsonic** [comment]
Sending up a PR to fix this now, thanks @jenniferlee-dsva! 🙂

**Comment by kelsonic** [comment]
PR is up! 🙂 https://github.com/department-of-veterans-affairs/vets-website/pull/15636

**Comment by jenniferlee-dsva** [comment]
I think I validated this already, so closing ticket. At least, I have that dejavu feeling.
